src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/WindowsForms/MyAppFileAccessor.cs

```diff
@@ -47,20 +65,34 @@ private void DocData_Modifying(object sender, EventArgs e)
 
     private async Task<MyAppDocument?> TryGetMyAppFileAsync()
     {
-        if (_docData is null)
+        if (_docData?.Data is null || !File.Exists(_absolutePath))
         {
-            string absolutePath = _project.MakeRooted("My Project\\Application.myapp");
-
             await _threadingService.SwitchToUIThread();
             try
             {
-                _docData = new DocData(_serviceProvider, absolutePath);
+                _docData = new DocData(_serviceProvider, _absolutePath);
                 _docData.Modifying += DocData_Modifying;
             } 
             catch (Exception)
             {
-                // If we have reached here, it means that the file doesn't exist in that location.
-                return null;
+                // If we have reached here, it means that the file doesn't exist in that location, so try to create it
+                
+                // Create My Project directory if it doesn't exist. If it does, nothing happens
+                await _storage.CreateFolderAsync("My Project");
+                
+                // Create and write defaults to the myapp file
+                await _storage.CreateEmptyFileAsync(_absolutePath);
+                StreamWriter writer = File.AppendText(_absolutePath);
+                await writer.WriteAsync(DefaultMyappFileContents);
+                await writer.FlushAsync();
+                writer.Close();
+                
+                // Now that we've created the file, we can instantiate _docData and myAppDocument.
+                _docData = new DocData(_serviceProvider, _absolutePath);
+                _docData.Modifying += DocData_Modifying;
+
+                myAppDocument = new MyAppDocument(_docData);
+                return myAppDocument;
```

Comment:
I wonder if we could extract this, so we don't have to use the main thread to create the location folder, create the file, write the file, flush the writer, and close it.

Reply:
Refactored!